dome/install_ubuntu_src.md

```diff
@@ -0,0 +1,292 @@
+sudo apt-get install git
+```
+
+## Getting the sources
+
+The instructions below use some UNIX commands to manage directories but the
+equivalent alternatives on Windows should provide the same result.
+
+The first step is to create a developer workspace in which `vcstool` and
+`colcon` can work:
+
+```bash
+mkdir -p ~/workspace/src
+cd ~/workspace/src
+```
+
+All the sources of ignition-dome are declared in a yaml file. Download
+it to the workspace:
+
+```bash
+wget https://raw.githubusercontent.com/ignition-tooling/gazebodistro/master/collection-dome.yaml
+```
+
+Use `vcstool` to automatically retrieve all the Ignition libraries sources from
+their repositories:
+
+```bash
+vcs import < collection-dome.yaml
+```
+
+The src subdirectory should contain all the sources ready to be built.
+
+## Install dependencies
+
+Before compiling it is necessary to install all the dependencies of the different
+packages that compose the Dome collection. Every platform has a different
+method to install software dependencies.
+
+Add `packages.osrfoundation.org` to the apt sources list:
+
+```bash
+sudo sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list'
+wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
+sudo apt-get update
+```
+
+The command below will install all dependencies in Ubuntu Focal:
+
+```bash
+sudo apt-get install -y \
+  cmake \
+  freeglut3-dev \
+  libavcodec-dev \
+  libavdevice-dev \
+  libavformat-dev \
+  libavutil-dev \
+  libbenchmark-dev \
+  libcurl4-openssl-dev \
+  libdart-collision-ode-dev \
+  libdart-dev \
+  libdart-utils-urdf-dev \
+  libfreeimage-dev \
+  libgflags-dev \
+  libglew-dev \
+  libgts-dev \
+  libjsoncpp-dev \
+  libogre-1.9-dev \
+  libogre-2.1-dev \
+  libprotobuf-dev \
+  libprotobuf-dev \
+  libprotoc-dev \
+  libqt5core5a \
+  libsqlite3-dev \
+  libswscale-dev \
+  libtinyxml-dev \
+  libtinyxml2-dev \
+  libwebsockets-dev \
+  libyaml-dev \
+  libzip-dev \
+  libzmq3-dev \
+  pkg-config \
+  protobuf-compiler \
+  python \
+  qml-module-qt-labs-folderlistmodel \
+  qml-module-qt-labs-settings \
+  qml-module-qtcharts \
+  qml-module-qtqml-models2 \
+  qml-module-qtquick-controls \
+  qml-module-qtquick-controls2 \
+  qml-module-qtquick-dialogs \
+  qml-module-qtquick-layouts \
+  qml-module-qtquick2 \
+  qtbase5-dev \
+  qtdeclarative5-dev \
+  qtquickcontrols2-5-dev \
+  ruby \
+  ruby-dev \
+  ruby-ronn \
+  swig \
+  uuid-dev
+```
+
+### Install compiler requirements
+
+The Ignition Libraries require the gcc 8 compiler on Ubuntu Focal.
+(Windows requires Visual Studio 2019).
+
+To install `gcc` version 8 on Ubuntu Focal:
+
+```bash
+sudo apt-get install g++-8
+```
+
+Set `gcc-8` and `g++-8` to be the default compilers.
+
+```bash
+sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-8 800 --slave /usr/bin/g++ g++ /usr/bin/g++-8 --slave /usr/bin/gcov gcov /usr/bin/gcov-8
```

Comment:
For what it is worth, I was able to compile on Focal with gcc/g++-9 (the default).  There were a lot of warnings and other stuff, but it did compile and run basic stuff for me.  It might be worth revisiting this and seeing if we can use the default instead.